### DISCUSSION TITLE
(node:20684) Error on developement env

### DISCUSSION THREAD

**Discussion by Manikanta-20** [vercel/next.js#14318] [Help]
(node:20684) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1)

some time i'm getting the above error
what is the solution?

**Comment by jamesmosier** [comment]
It is hard to say without seeing some code. This most often means that an uncaught error occurred within `getServerSideProps`, `getStaticProps`, or `getStaticPaths`. Do you have any code you could share?

**Reply by Manikanta-20** [reply]
@jamesmosier  Please verify the below code. This issue will come some time

import React from 'react';
import { useRouter } from 'next/router';
import Typography from '@material-ui/core/Typography';
import TableContainer from '@material-ui/core/TableContainer';
import TableBody from '@material-ui/core/TableBody';
import TableRow from '@material-ui/core/TableRow';
import TableCell from '@material-ui/core/TableCell';
import { withStyles, makeStyles } from '@material-ui/core/styles';
import FormControlLabel from '@material-ui/core/FormControlLabel';

import Checkbox from '@material-ui/core/Checkbox';
import Link from '@material-ui/core/Link';
import { Box } from '@material-ui/core';
import Button from '@material-ui/core/Button';

import { Layout } from '../app/components/layout';
import useSnackBars from '../app/components/toast/toast';

export const StyledCell = withStyles({
  root: {
    border: 'unset',
  },
})(TableCell);
const useStyles = makeStyles(() => ({
  body: {
    display: 'flex',
    justifyContent: 'center',
    alignSelf: 'center',
  },
}));
const InfoPage = () => {
  const router = useRouter();
  const [checked, setState] = React.useState(false);
  const { addAlert } = useSnackBars();
  const classes = useStyles();
  const handleChange = (event) => {
    setState(event.target.checked);
  };
const preventDefault = (event) => event.preventDefault();
  const handleClick = (event) => {
    event.preventDefault();

    if (checked) {
      router.push({
        pathname: '/finalstep2',
        query: {
          name: router.query.name,
          address: router.query.address,
          id: router.query.id,
        },
      });
    } else {
      addAlert({ status: 'warning', message: 'Please check terms and conditions' });
    }
  };
  return (
    <Layout>
      <Typography component="h5" variant="h5" align="center">
        Infomation
      </Typography>

      <hr width="50%" align="center" style={{ marginTop: '8px' }} />
      <TableContainer component="div" className={classes.body}>
        <TableBody>
          <TableRow style={{ border: 'unset' }}>
            <StyledCell component="th" scope="row">
              <Typography component="span">Name</Typography>
            </StyledCell>
            <StyledCell component="th" scope="row">
              <Typography component="span">{router.query.name}</Typography>
            </StyledCell>
          </TableRow>
          <TableRow>
            <StyledCell component="th" scope="row">
              <Typography component="span">Address</Typography>
            </StyledCell>
            <StyledCell component="th" scope="row">
              <Typography component="span"> {router.query.address}</Typography>
            </StyledCell>
          </TableRow>
        </TableBody>
      </TableContainer>
      <FormControlLabel
        className={classes.body}
        control={
          <Checkbox
            color="primary"
            checked={checked}
            onChange={handleChange}
            name="term-condition"
          />
        }
        label={
          <Link href="#" onClick={preventDefault}>
            I agree to terms and conditions
          </Link>
        }
      />
      <Box component="div" justifyContent="center" display="flex">
        <Button color="primary" variant="contained" onClick={handleClick}>
          Submit
        </Button>
        <span style={{ width: '20px' }} />
        <Button variant="contained" onClick={() => router.back()}>
          Cancel
        </Button>
      </Box>
    </Layout>
  );
};

export default InfoPage;

Same issue i got for another page that contains only view part.

**Reply by Manikanta-20** [reply]
@jamesmosier  any update?

**Reply by jamesmosier** [reply]
i don't see anything wrong with the above code that would cause that, so it is likely elsewhere in your application

**Reply by Manikanta-20** [reply]
@jamesmosier  some time will get this error ,it will  block complete application.you cannot navigate  any page . Error number(node:xxxxxx)  will be depends node version

**Reply by Manikanta-20** [reply]
@jamesmosier did you find any find solution for this?

**Reply by jamesmosier** [reply]
if you have a full reproduction with all code i can certainly take a look.
